framework/src/wifi_manager/virtual_wifi_utils.c

```diff
@@ -0,0 +1,92 @@
+/****************************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <sys/types.h>
+#include <debug.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include "wifi_utils.h"
+
+
+wifi_utils_result_e wifi_utils_init(void)
```

Comment:
1. This looks not a part of `Add the configuration which supports network binary separation`.
Please split a commit.
2. Do you need to make dummy functions? How about modify as shown below in wifi_utils.h?
```
#ifdef CONFIG_SELECT_NO_DRIVER
#define wifi_utils_init
#define BBBB
#define CCC
#else 
wifi_utils_result_e wifi_utils_init(void);
wifi_utils_result_e wifi_utils_deinit(void);
....
#endif 
```
This can reduce function calls with CONFIG_SELECT_NO_DRIVER.

Reply:
Thanks for your comment but I prefer adding virtual utils.